### PR TITLE
docs: match docs for "network upgrade"

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

Re-defines a hardfork as a "breaking network upgrade" and swaps out "hardfork" for network upgrade, to be more generalizable.

### CODE REVIEW

**Review by rootulp — APPROVED** [review]
We may need to also change it in the table of contents: https://github.com/celestiaorg/CIPs/blob/389ed243c44bf93b7024240c2c089afb34d21662/cips/README.md?plain=1#L47

**Comment by jcstein** [comment]
I think I got everything in repo in https://github.com/celestiaorg/CIPs/pull/205/commits/0d6db450337b1425b66da20933833ff367bbdbbe

File: cips/notes/cdc-15.md [hunk 1]
```diff
@@ -25,7 +25,7 @@
 *   Celestia Node to release update supporting Celestia App v2 soon
 *   Arabica devnet upgrade planned for **August 14th**
 *   Node operators must upgrade within **1 week** timeframe
-*   Consensus node operators need to use a specific CLI flag ([see documentation](https://docs.celestia.org/nodes/hardfork-process#lemongrass-hardfork))
+*   Consensus node operators need to use a specific CLI flag ([see documentation](https://docs.celestia.org/nodes/network-upgrade-process#lemongrass-network-upgrade))
```
Comment (jcstein) [inline cips/notes/cdc-15.md:28]:
```suggestion
*   Consensus node operators need to use a specific CLI flag ([see documentation](https://docs.celestia.org/nodes/network-upgrade-process#lemongrass-network-upgrade))
```

Comment (jcstein) [inline cips/notes/cdc-15.md:28]: @rootulp I realized that I made the link "network upgrades" on accident due to the heading. so have fixed in: https://github.com/celestiaorg/docs/pull/1714

**Pull request merged**